www/js/init.js

```diff
@@ -30,6 +30,13 @@
  */
 var params = {};
 
+/**
+ * A global object for storing app state
+ * 
+ * @type Object
+ */
+var globalstate = {};
```

Comment:
This variable can probably be moved to app.js, but it's minor and can be done later

Reply:
Thanks @mossroy . It would feel safer to me to move this as part of #637 when I'll have time to test a unified solution to cancelling running binary searches (and possibly also cancelling fetching of assets on pages we have navigated away from). So I'll move a summary of issues to be dealt with to #637 after squashing and merging this PR.

Reply:
Sure, that's perfect